### PR TITLE
Add two more plotters for resolution

### PR DESCRIPTION
**Pull request by benkrikler**
This PR adds:
 * Recalc methods to deduce the resolution (difference between online and offline) for different quantities.
 * A plotter to plot the resolution distribution vs pile-up
 * A plotter to plot the resolution distributions against another variable (eg. eta or phi) in pile-up bins

### CODE REVIEW

File: cmsl1t/plotting/resolution_vs_X.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+            # Save canvas to file
+            name = self.filename_format.format(pileup=pileup)
+            self.save_canvas(canvas, name + suffix)
+
+    def _is_consistent(self, new):
+        """
+        Check the two plotters are the consistent, so same binning and same axis names
+        """
+        return (self.pileup_bins.bins == new.pileup_bins.bins) and \
```
Comment (kreczko) [inline cmsl1t/plotting/resolution_vs_X.py]: tempted to modify this for readability:
```python
attrsToCompare =['resolution_method', 'versus_name', 'online_name', 'offline_name']
results = [getattr(self, attr) == getattr(new, attr) for atttr in attrsToCompare]
results.append(self.pileup_bins.bins == new.pileup_bins.bins)
return all(results)
```

On second thought, maybe better to
```python
return all ([
    self.pileup_bins.bins == new.pileup_bins.bins,
    self.resolution_method == new.resolution_method,
    self.versus_name == new.versus_name,
    self.online_name == new.online_name,
    self.offline_name == new.offline_name
])
```
(fewer 'ands')

File: cmsl1t/plotting/resolution_vs_X.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            canvas = draw2D(hist, draw_args={"xtitle": self.versus_title, "ytitle": ytitle})
+
+            # Add labels
+            label_canvas()
+
+            # Save canvas to file
+            name = self.filename_format.format(pileup=pileup)
+            self.save_canvas(canvas, name + suffix)
+
+    def _is_consistent(self, new):
```
Comment (kreczko) [inline cmsl1t/plotting/resolution_vs_X.py]: Isn't this implementation identical to `ResolutionPlot`?

Comment (benkrikler) [reply cmsl1t/plotting/resolution_vs_X.py]: No, it checks the `versus_name` variable as well.

File: cmsl1t/plotting/resolution_vs_X.py (hunk 1 above)
Comment (benkrikler) [reply cmsl1t/plotting/resolution_vs_X.py]: I prefer using `all` the most.  The first option feels actually less clear.  An alternative, for the future maybe, is to have each plotter specify which attributes need to be checked and have the base class actually implement the checking of these.

**Pull request merged**